src/esportsbot/cogs/DefaultRoleCog.py

```diff
@@ -11,88 +11,81 @@ def __init__(self, bot):
 
     @commands.Cog.listener()
     async def on_member_join(self, member):
-        guild = DBGatewayActions().get(Guild_info, guild_id=member.guild.id)
-        if not guild:
-            return
-
-        if guild.default_role_id:
-            default_role = member.guild.get_role(guild.default_role_id)
-            await member.add_roles(default_role)
-            await send_to_log_channel(
-                self,
-                member.guild.id,
-                f"{member.mention} has joined the server and received the {default_role.mention} role"
-            )
-        else:
-            await send_to_log_channel(self, member.guild.id, f"{member.mention} has joined the server")
+        # Get all the default role for the server from database
+        guild_default_roles = DBGatewayActions().list(Default_roles, guild_id=member.guild.id)
+        # Check to see if any roles exist
+        if guild_default_roles:
+            # Create list of roles from database response
+            apply_roles = [member.guild.get_role(role.role_id) for role in guild_default_roles]
+            # Add all the roles to the user, we don't check if they're valid as we do this on input
+            await member.add_roles(*apply_roles)
 
     @commands.command(
-        name="setdefaultrole",
-        usage="<role_id> or <@role>",
-        help="Sets the role that the server gives to members when they join the server"
+        name="setdefaultroles",
+        usage="<@role> <@role> <@role> ...",
+        help="Sets the roles that the server gives to members when they join the server"
     )
     @commands.has_permissions(administrator=True)
-    async def setdefaultrole(self, ctx, given_role_id):
-        cleaned_role_id = role_id_from_mention(given_role_id) if given_role_id else False
-        if cleaned_role_id:
-            guild = DBGatewayActions().get(Guild_info, guild_id=ctx.author.guild.id)
-
-            if not guild:
-                db_item = Guild_info(guild_id=ctx.guild.id, default_role_id=cleaned_role_id)
-                DBGatewayActions().create(db_item)
-            else:
-                guild.default_role_id = cleaned_role_id
-                DBGatewayActions().update(guild)
-
-            await ctx.channel.send(self.STRINGS['default_role_set'].format(role_id=cleaned_role_id))
-            default_role = ctx.author.guild.get_role(cleaned_role_id)
-            await send_to_log_channel(
-                self,
-                ctx.author.guild.id,
-                self.STRINGS['default_role_set_log'].format(author=ctx.author.mention,
-                                                            role_mention=default_role.mention)
-            )
+    async def setdefaultroles(self, ctx, *, args: str):
```

Comment:
I mentioned it in the previous PR, but just to be thorough will bring it up here: You could make use of the `discord.Context.message.role_metions` attribute, as that will give you all the roles mentioned in the message(command) as role objects

Reply:
See https://github.com/FragSoc/esports-bot/issues/163